src/gtc/gtir_to_oir.py

```diff
@@ -230,10 +264,12 @@ def visit_VerticalLoop(self, node: gtir.VerticalLoop, *, ctx: Context) -> oir.Ve
     def visit_Stencil(self, node: gtir.Stencil) -> oir.Stencil:
         ctx = self.Context()
         vertical_loops = self.visit(node.vertical_loops, ctx=ctx)
-        return oir.Stencil(
-            name=node.name,
-            params=self.visit(node.params),
-            vertical_loops=vertical_loops,
-            declarations=ctx.temp_fields,
-            loc=node.loc,
+        return validate_stencil_memory_accesses(
```

Comment:
The place looks a bit weird. Maybe we can run this after the lowering, e.g. place it in the pipeline after the lowering. Or temporarily as a root_validator in the Stencil (temporarily because when we switch to the updated eve, we probably can not).

Reply:
I originally had it in `oir_pipeline.DefaultPipeline` as the first pass, but on second thought realized that passes should be reserved for optimizations, not correctness checks. What do you think?